sympy/sets/sets.py

```diff
@@ -729,9 +729,10 @@ def __new__(cls, start, end, left_open=False, right_open=False):
             raise ValueError("Non-real intervals are not supported")
 
         # evaluate if possible
-        if (end < start) == True:
+        is_empty = (end - start).is_negative
+        if (end < start) is True:
```

Comment:
This should probably have `==` instead of `is` if  `end` or `start` is a SymPy object. For example,
```
In [21]: (0 < sqrt(2)) is True
Out[21]: False

In [22]: (0 < sqrt(2)) == True
Out[22]: True
```
Even `if end < start:` could suffice.